app/src/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -1837,11 +1838,13 @@ public void handleFormLoadCompletion(FormController fc) {
      * Call when the user provides input that they want to quit the form
      */
     private void triggerUserQuitInput() {
-        //If we're just reviewing a read only form, don't worry about saving
-        //or what not, just quit
-        if(mFormController.isFormReadOnly()) {
-            //It's possible we just want to "finish" here, but
-            //I don't really wanna break any c compatibility
+        if(!formHasLoaded()) {
+            finish();
```

Comment:
@phillipm Going to merge this PR, but would be good to get a quick explanation of what this is doing-- seems unrelated to the rest of the PR.